x/utils/httpQueries.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+//queryFromAPI is generic function that can query every api and endpoint
+func queryFromAPI(url string, endpoint string, ptr interface{}) error {
```

Comment:
I think that having the `url` and `endpoint` separated is not the best way to ask users for such parameters. Instead, it would be much easier for them to speify a single `endpoint` parameter that already contains the `url` part itself.